### PR TITLE
Create feeding-data-amp-pages-graphql.liquid

### PR DESCRIPTION
**Pull request by diana-lakatos**
Feeding Data into AMP Pages with GraphQL tutorial for the How-To Guides/AMP section. 3/3 of AMP series.

### CODE REVIEW

**Comment by pavelloz** [comment]
I would add how application layout looks like - it has to have special character in the html tag to make it valid AMP Page

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+Create the main page, HTML extended with AMP tags and objects `marketplace_builder/views/pages/product-list.liquid`:
+
+<pre><code class="language-liquid">
+---
+slug: product-list
+format: html
+layout_name: application
+---
+<!-- AMP dependencies -->
+<script custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js" async=""></script>
```
Comment (pavelloz) [inline marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: Script tags need to be in content_for and this content_for needs to be yielded in the layout. Its not good to put javascripts in the middle of the page.

Something like this:
https://github.com/mdyd-dev/amp-demo-pages/blob/master/travel/marketplace_builder/liquid_views/layouts/application.liquid#L17 +
https://github.com/mdyd-dev/amp-demo-pages/blob/master/travel/marketplace_builder/pages/travel.liquid#L7-L11

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+* The data source for `<amp-list>` is either `products-asc.json` or `products-desc.json` and is chosen based on the state variable `products.order`.
+* `amp-mustache` enables the 'mustache' (double curly braces) templating syntax.
+* The `{% raw %}` tag instructs the liquid preprocessor not to interpret the `{{}}` tags, and leaves them to AMP/mustache to process. 
+
+
+### Step 2: Create JSON endpoint 
+Create the JSON endpoint, a dynamically generated JSON document from which AMP fetches the data that it displays `mviews/pages/api/products-asc.json.liquid`: 
+
+<pre><code class="language-liquid">
+---
```
Comment (pavelloz) [inline marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: raw / endraw needed for it to display in docs

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  - field_name: price
+    required: true
+    validation_only_on_update: false
+</code></pre>
+
+### Step 3: Display product details
+Create the `/product-details` page that loads when visitors click on a product from `/product-list` above `views/pages/product-details.liquid`:
+
+<pre><code class="language-liquid">
+---
```
Comment (pavelloz) [inline marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: raw / endraw

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  html_tag: input
+  input_html_options: {}
+  label: Price
+  public: true
+  search_in_query: false
+  searchable: false
+  validation:
+  - field_name: price
+    required: true
+    validation_only_on_update: false
```
Comment (pavelloz) [inline marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: From all above this is sufficient:

```
custom_attributes:
- name: price
  attribute_type: float
```

**Comment by pavelloz** [comment]
Is it deployed somewhere where i can look at it?

Im pretty sure that creating `api/products.json` and asking `api/products-desc.json`/`api/products-asc.json` wont work.

**Review by pavelloz — CHANGES_REQUESTED** [review]

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid (hunk 2 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: Thanks! Fixed.

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid (hunk 3 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: Thanks! Fixed.

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid (hunk 4 above)
Comment (diana-lakatos) [reply marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: Thank you! Fixed.

File: marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid (hunk 1 above)
Comment (ghost) [reply marketplace_builder/views/pages/how-to/amp/feeding-data-amp-pages-graphql.liquid]: @pavelloz  You are right, I was just trying to keep it short. Changed now. (Lucian)

**Comment by ghost** [comment]
@pavelloz @diana-lakatos Removed remnants of product sorting. (Lucian)

**Review by pavelloz — APPROVED** [review]

**Comment by pavelloz** [comment]
Thanks :)

**Pull request merged**